packages/editor/src/browser/editor-manager.ts

```diff
@@ -112,14 +114,38 @@ export class EditorManagerImpl implements EditorManager, WidgetFactory {
     }
 
     protected createEditor(uri: URI): Promise<EditorWidget> {
-        return this.editorProvider(uri).then(textEditor => {
-            const newEditor = new EditorWidget(textEditor, this.selectionService);
-            newEditor.id = this.id + ":" + uri.toString();
-            newEditor.title.closable = true;
-            newEditor.title.label = uri.path.base;
-            newEditor.title.iconClass = this.iconProvider.getFileIconForURI(uri);
-            return newEditor;
-        });
+        const editorPromise = this.editorProvider(uri);
+        const workspaceRootPromise = this.workspaceServie.root;
+
+        return Promise.all([editorPromise, workspaceRootPromise])
+            .then(res => {
+                const [textEditor, workspaceRoot] = res;
+
+                const newEditor = new EditorWidget(textEditor, this.selectionService);
+                newEditor.id = this.id + ":" + uri.toString();
+                newEditor.title.closable = true;
+
+                /* Put just the base filename in the tab itself.  */
+                newEditor.title.label = uri.path.base;
+
+                /* Put the workspace-relative path in the caption (hover popup).  */
+                let caption = uri.path.toString();
```

Comment:
looks like a new method `getCaption(URI)` missing, an inline comment can go to doc-string

Reply:
it looks bogus:
- the caption could be `uri.path` only if the uri is `FileUri`, otherwise it should be `uri.toString()`. I actually would prefer just always `uri.toString()`, since even `FileUri` can have other important parts except `path`, like `query`
- you should compare URIs, not paths to get a relative workspace URI regardless to schema

Reply:
Please let's not simply display the URI. It is more like a technical, internal representation.

We already have a couple of places, where we need to display a long version of a URI (problems view, open resource, git view). It is time to generalize that to something like a LabelProvider for URIs. I think it would be worth applying a contribution point, similar to how we do the OpenHandler stuff. The default implementation should use `${displayName} - ${parentPathWithoutWSRoot}`.

Reply:
I am working on moving the icon provider to core and will introduce a common label provider, too.